tests/ui/lint/int_to_ptr.stderr

```diff
@@ -0,0 +1,71 @@
+warning: transmuting an integer to a pointer without provenance is dangerous
+  --> $DIR/int_to_ptr.rs:10:25
+   |
+LL |     let _ptr = unsafe { std::mem::transmute::<usize, *const u8>(a) };
+   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: dereferencing the resulting pointer is undefined behavior as it doesn't have any provenance
+   = note: exposed provenance semantics can be used to create a pointer based on some previously exposed provenance
+   = help: for more information about transmute, see <https://doc.rust-lang.org/std/mem/fn.transmute.html#transmutation-between-pointers-and-integers>
+   = help: for more information about exposed provenance, see <https://doc.rust-lang.org/std/ptr/index.html#exposed-provenance>
+   = help: use `std::ptr::without_provenance_mut` to create a pointer without provenance
```

Comment:
```suggestion
   = help: if you truly mean to create a pointer without provenance, use `std::ptr::without_provenance_mut`
```

Also in terms of ordering, I think this might better be put above the URLs